node/src/proxy_server/mod.rs

```diff
@@ -162,12 +164,35 @@ impl AddReturnRouteMessage {
     }
 }
 
-impl Handler<AddRouteMessage> for ProxyServer {
+impl Handler<RouteResultMessage> for ProxyServer {
     type Result = ();
 
-    fn handle(&mut self, msg: AddRouteMessage, _ctx: &mut Self::Context) -> Self::Result {
-        debug!(self.logger, "Establishing stream key {}", msg.stream_key);
-        self.stream_key_routes.insert(msg.stream_key, msg.route);
+    fn handle(&mut self, msg: RouteResultMessage, _ctx: &mut Self::Context) -> Self::Result {
+        let dns_failure = self
+            .dns_failure_retries
+            .get(&msg.stream_key)
+            .unwrap_or_else(|| {
+                panic!("RouteResultMessage Handler: stream key: {} not found within dns_failure_retries", msg.stream_key);
+            });
+
+        match msg.result {
+            Ok(route_query_response) => {
+                debug!(
+                    self.logger,
+                    "Found a new route for hostname: {:?} - stream key: {}  retries left: {}",
+                    dns_failure.unsuccessful_request.target_hostname,
+                    msg.stream_key,
+                    dns_failure.retries_left
+                );
+                self.stream_key_routes
+                    .insert(msg.stream_key, route_query_response);
+                // TODO Preserve all the previously know routes and halt the retry process if the new route is already present.
```

Comment:
Has this `// TODO` been done? If so, remove the comment...

Reply:
This TODO was just an inspiration for a new card. It's not something that should be within the scope of the current card. It'll be appropriate if we have a discussion whether this comment is worth a new card or it'll be more appropriate to discard this idea and remove this TODO.

Reply:
@dnwiebe